src/snowflake/connector/connection.py

```diff
@@ -276,8 +281,18 @@ class SnowflakeConnection:
         schema: Schema in use on Snowflake.
         warehouse: Warehouse to be used on Snowflake.
         role: Role in use on Snowflake.
-        login_timeout: Login timeout in seconds. Used while authenticating.
-        network_timeout: Network timeout. Used for general purpose.
+        login_timeout: Login timeout in seconds. Login requests will not be retried after this timeout expires.
+            Note that the login attempt may still take more than login_timeout seconds as an ongoing login request
+            cannot be canceled even upon login timeout expiry. The login timeout only prevents further retries.
+            If not specified, login_timeout is set to DEFAULT_AUTH_CLASS_TIMEOUT in the by_plugin module.
```

Comment:
nit: let's be clear where the default value is
```suggestion
            If not specified, login_timeout is set to `snowflake.connector.auth.by_plugin.DEFAULT_AUTH_CLASS_TIMEOUT`.
```